app.py

```diff
@@ -151,100 +169,77 @@ async def delete_user(user_id):
                 await cur.execute("DELETE FROM users WHERE user_id = %s", (user_id,))
                 await cur.close()
                 pool.close()
-                return jsonify({'message': 'User deleted successfully!'}), 200
+                return jsonify({"message": "User deleted successfully!"}), 200
 
 
 # Obtain token access and refresh pairs endpoint
-@app.post('/login')
+@app.post("/login")
 async def login():
-    user_name = request.json['user_name']
-    password = request.json['password']
+    user_name = request.json["user_name"]
+    password = request.json["password"]
     user = await check_user_exist(name=user_name)
-    if not user or not check_password_hash(user[3], password):
-        return jsonify({'error': 'Invalid username or password.'}), 401
-    access_token = jwt.encode({'user_id': user[0], 'exp': datetime.datetime.utcnow() + datetime.timedelta(minutes=60)},
-                              app.config['SECRET_KEY'],
-                              algorithm='HS256')
-    refresh_token = jwt.encode({'user_id': user[0], 'exp': datetime.datetime.utcnow() + datetime.timedelta(days=7)},
-                               app.config['SECRET_KEY'], algorithm='HS256')
-    access_expire = datetime.datetime.utcnow() + datetime.timedelta(minutes=60)
-    refresh_expire = datetime.datetime.utcnow() + datetime.timedelta(days=7)
-    pool = await create_pool()
-    async with pool.acquire() as conn:
-        async with conn.cursor() as cur:
-            await cur.execute('INSERT INTO refresh_tokens (user_id, refresh_token) VALUES (%s, %s)',
-                              (user[0], refresh_token))
-            await cur.close()
-            pool.close()
-    return jsonify({'access_token': access_token, 'refresh_token': refresh_token, 'access_expire': access_expire,
-                    'refresh_expire': refresh_expire}), 200
+    user_dict = dict(zip(("user_id", "user_name", "email", "password"), user))
+    if not user or not check_password_hash(user_dict["password"], password):
+        return jsonify({"error": "Invalid username or password."}), 401
+    return create_and_return_token(user_dict["user_id"])
 
 
 # Obtain new access token from refresh token endpoint
-@app.route('/refresh', methods=['POST'])
+@app.post("/refresh")
 async def refresh():
-    refresh_token = request.json['refresh_token']
-    pool = await create_pool()
-    async with pool.acquire() as conn:
-        async with conn.cursor() as cur:
-            await cur.execute('SELECT * FROM refresh_tokens WHERE refresh_token = %s', (refresh_token,))
-            token = await cur.fetchone()
-            await cur.close()
-            pool.close()
-    if not token:
-        return jsonify({'error': 'Invalid refresh token.'}), 401
+    refresh_token = request.json["refresh_token"]
```

Comment:
When you refresh the token following issues can be occured.

1. User may send an access token and get a new refresh token. You don't check if it's a refresh token or an access token. To do that, you may have to specify the token type.
2. You don't check the DB for the particular user. What if that user is recently deleted? But still you send valid refresh tokens.